grep-0001-coding-guidelines.md

```diff
@@ -73,6 +72,20 @@ and follow them.
   can thus be optimized better at compile time. In many cases, lambdas are
   also more readable than the `bind` calls, which have add their own
   syntactical idiosyncrasies (such as the `_1` argument etc.).
+* Feel free to use modern C/C++ features even if they were not used before.
+  Make sure they work with the compilers and dependencies which are set for the
+  version of GNU Radio the commit will be made upon. The GNU Radio CI system
+  will be able to confirm this. (Note: C++11 features are available starting
+  with GNU Radio version 3.8).
+
+#### Code Style Guidelines
+* Use the [LLVM code formatting guide](https://llvm.org/docs/CodingStandards.html#source-code-formatting). Following points will specify differences to it and highlight important points.
+* Do not indent namespaces
+* Use 4 spaces indentation width for new code.
```

Comment:
I like 4 spaces, but none of our code does that. This is a big change, it'll affect future diffs etc.

Reply:
I talked with @marcusmueller about that, and looking through our source code we don't have a clear cut on either side. It's all mixed up, even in single files.

Reply:
Yeah, and at least we'll finally have an actual guideline. Like I said, it's probably worth auto-reformatting the source tree at one point, after that, it'll be much easier to encourage these rules properly.